File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/DiscoveringOriginService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.DiscoveringOrigin;

public interface DiscoveringOriginService {
    public List<DiscoveringOrigin> getDiscoveringOrigins();

    public DiscoveringOrigin getDiscoveringOrigin(Integer id);

    public DiscoveringOrigin addOrUpdateDiscoveringOrigin(DiscoveringOrigin discoveringOrigin);
}

File: client/src/app/modules/tiers/services/discovering.origin.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { DiscoveringOrigin } from '../../tiers/model/DiscoveringOrigin';

@Injectable({
  providedIn: 'root'
})
export class DiscoveringOriginService extends AppRestService<DiscoveringOrigin>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getDiscoveringOrigins() {
    return this.getList(new HttpParams(), "dicovering-origins");
  }
  
   addOrUpdateDiscoveringOrigin(discoveringOrigin: DiscoveringOrigin) {
    return this.addOrUpdate(new HttpParams(), "discovering-origin", discoveringOrigin, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
